web/app/scripts/src/directives/form.es.js

```diff
@@ -39,11 +50,40 @@ module.directive('surveysForm', [
                             // been set.
                             if (Object.keys(scope.model).length === 0) {
                                 scope.model = FieldsetParser.parseFormToModel(structure);
+                                countQuestionsTotal();
                             }
                         });
                     }
                 });
 
+                // Check and count the number of questions answered.
+                const checkAnswered = function (answers) {
```

Comment:
even if its clear code i like to put comments if someone who was new to javascript the comments will help make it easier to read